tools/PI/DevHome.PI/Helpers/ExternalToolsHelper.cs

```diff
@@ -15,14 +17,44 @@ namespace DevHome.PI.Helpers;
 internal sealed class ExternalToolsHelper
 {
     private readonly JsonSerializerOptions serializerOptions = new() { WriteIndented = true };
-    private readonly ObservableCollection<ExternalTool> externalTools = [];
     private readonly string toolInfoFileName;
 
     public static readonly ExternalToolsHelper Instance = new();
 
     private static readonly ILogger _log = Log.ForContext("SourceContext", nameof(ExternalToolsHelper));
 
-    public ReadOnlyObservableCollection<ExternalTool> ExternalTools { get; set; }
+    private ObservableCollection<ExternalTool> allExternalTools = [];
+
+    // The ExternalTools menu shows all registered tools.
+    public ObservableCollection<ExternalTool> AllExternalTools
+    {
+        get => allExternalTools;
+        set
+        {
+            // We're assigning the collection once, and this also covers the case where we reassign it again:
+            // we need to unsubscribe from the old collection's events, subscribe to the new collection's events,
+            // and initialize the filtered collection.
+            if (allExternalTools != value)
+            {
+                if (allExternalTools != null)
+                {
+                    allExternalTools.CollectionChanged -= AllExternalTools_CollectionChanged;
+                }
+
+                allExternalTools = value;
+                if (allExternalTools != null)
+                {
+                    allExternalTools.CollectionChanged += AllExternalTools_CollectionChanged;
+                }
+
+                // Synchronize the filtered collection with this unfiltered one.
+                SynchronizeAllFilteredItems();
+            }
+        }
+    }
+
+    // The bar shows only the pinned tools.
+    public ObservableCollection<ExternalTool> FilteredExternalTools { get; private set; } = [];

```

Comment:
>ObservableCollection [](http://example.com/codeflow?start=11&length=20)

Is this a ReadOnly collection?

Reply:
No, I thought about wrapping it, but the set is private, and I need to be able to change at least the IsPinned property on any of the items in the collection.

Reply:
ReadonlyCollection only prevents you from adding/removing from the collection, not modifying the items in the collection. I think you want this as a ReadOnly collection, as you don't want other components adding/removing from this collection, correct?

Reply:
Fixed in 3rd commit.